test/Esprima.Tests.Test262/Esprima.Tests.Test262.csproj

```diff
@@ -28,4 +29,16 @@
       <Content Include=".config\dotnet-tools.json" />
     </ItemGroup>
 
+    <!-- Based on the idea presented at https://mhut.ch/journal/2015/06/30/build-time-code-generation-in-msbuild -->
+    <Target Name="GenerateTestSuite" DependsOnTargets="_GenerateTestSuite" BeforeTargets="BeforeBuild;BeforeRebuild" Condition="!Exists($(GeneratedTestSuiteDir))">
```

Comment:
`test262 generate` would read `test262.state` file containing a generation hash and check that against current config file and probably current output files. If not matching, need to re-generate. Currently solution requires clean rebuild if for example current commit hash changes in file.

Reply:
Yeah, that would be nice indeed. However, as you mentioned, this would imply that the tool would need to run on every build, so in this case maybe it'd be better to implement it as a custom MSBuild task rather than a dotnet tool. AFAIK, MSBuild extensions (tasks, targets, etc.) can be published as NuGet packages as well.

But regardless the implementation details, there's definitely some room for improvement here. However, this would also involve touching your Test262 project, so it's kind of out-of-scope for this PR. This is a nice task for the future but I think for now that's already an improvement if developers don't have to generate manually and the solution compiles without errors after a clean checkout.

Reply:
Agree. I just wanted to add some context. Current solution you are building is a great improvement already.